crates/wysiwyg/src/dom/unicode_string.rs

```diff
@@ -59,6 +59,42 @@ pub trait UnicodeStr:
 
     // Should really be `-> Self::Chars<'a>`, but that requires GATs
     fn chars(&self) -> Box<dyn Iterator<Item = char> + '_>;
+
+    /// Returns the length of the char in indices of the current encoding
+    fn char_len(&self, char: &char) -> usize;
+
+    /// Splits the current UnicodeStr into graphemes (visual characters), from start to end indexes
+    fn graphemes(&self, start: usize, end: usize) -> Vec<&Self> {
+        let mut ret = Vec::new();
+        let mut start = start;
+        let mut offset = 0;
+        let mut chars = self.chars().peekable();
+        while chars.peek().is_some() {
+            let c = chars.next().unwrap();
+            let char_len = self.char_len(&c);
+            let new_offset = offset + char_len;
```

Comment:
I think the code works (I tried to break it with a test and couldn't), but I can't understand why we don't need `new_offset += char_len;` here. Surely every time around the loop we should move further to the right?

Reply:
Ah, I found where we update `offset` now, ignore me!